storage/src/test/java/org/apache/kafka/server/log/remote/metadata/storage/TopicBasedRemoteLogMetadataManagerTest.java

```diff
@@ -16,153 +16,128 @@
 import org.apache.kafka.server.log.remote.storage.RemoteResourceNotFoundException;
 import org.apache.kafka.server.log.remote.storage.RemoteStorageException;
-import org.junit.jupiter.api.AfterEach;
 import org.junit.jupiter.api.Assertions;
-import org.junit.jupiter.api.BeforeEach;
-import org.junit.jupiter.api.Test;
+import org.junit.jupiter.api.extension.ExtendWith;
 import org.slf4j.Logger;
 import org.slf4j.LoggerFactory;
-import scala.collection.JavaConverters;
-import scala.collection.Seq;
 
 import java.io.IOException;
-import java.util.ArrayList;
+import java.util.Arrays;
 import java.util.Collections;
-import java.util.HashMap;
-import java.util.List;
-import java.util.Properties;
 import java.util.concurrent.TimeoutException;
 
-@SuppressWarnings("deprecation") // Added for Scala 2.12 compatibility for usages of JavaConverters
+@ExtendWith(ClusterTestExtensions.class)
+@ClusterTestDefaults(brokers = 3, clusterType = Type.ALL)
 public class TopicBasedRemoteLogMetadataManagerTest {
     private static final Logger log = LoggerFactory.getLogger(TopicBasedRemoteLogMetadataManagerTest.class);
 
     private static final int SEG_SIZE = 1024 * 1024;
 
     private final Time time = new MockTime(1);
-    private final TopicBasedRemoteLogMetadataManagerHarness remoteLogMetadataManagerHarness = new TopicBasedRemoteLogMetadataManagerHarness();
 
-    @BeforeEach
-    public void setup() {
-        // Start the cluster and initialize TopicBasedRemoteLogMetadataManager.
-        remoteLogMetadataManagerHarness.initialize(Collections.emptySet(), true);
-    }
-
-    @AfterEach
-    public void teardown() throws IOException {
-        remoteLogMetadataManagerHarness.close();
-    }
-
-    public TopicBasedRemoteLogMetadataManager topicBasedRlmm() {
-        return remoteLogMetadataManagerHarness.remoteLogMetadataManager();
-    }
-
-    @Test
-    public void testDoesTopicExist() {
-        Properties adminConfig = remoteLogMetadataManagerHarness.adminClientConfig();
-        ListenerName listenerName = remoteLogMetadataManagerHarness.listenerName();
-        try (Admin admin = remoteLogMetadataManagerHarness.createAdminClient(listenerName, adminConfig)) {
+    @ClusterTest
+    public void testDoesTopicExist(ClusterInstance clusterInstance) throws Exception {
+        try (TopicBasedRemoteLogMetadataManager remoteLogMetadataManager = RemoteLogMetadataManagerTestUtils.builder()
+                .bootstrapServers(clusterInstance.bootstrapServers()).startConsumerThread(true).build();
+             Admin admin = clusterInstance.createAdminClient()) {
             String topic = "test-topic-exist";
-            remoteLogMetadataManagerHarness.createTopic(topic, 1, 1, new Properties(),
-                    listenerName, adminConfig);
-            boolean doesTopicExist = topicBasedRlmm().doesTopicExist(admin, topic);
+            admin.createTopics(Collections.singletonList(new NewTopic(topic, 1, (short) 1))).all().get();
```

Comment:
We used to use `TestUtils.createTopicWithAdmin` to make sure the topic is created and all metadata are propagated. I think we should keep using it, to make the test reliable.

Same comments to below createTopic places.